### PR TITLE
Fix issues caused by updating to Factorio 2.0

### PR DESCRIPTION
**Pull request by mGuv**
I've updated all the errors I could find.

- Fixed differences in Factorio 2.0 modding API
- Updated to a 2.0 version of stdlib and flib
- Copy pasted in an old version of flib.gui so I didn't have to rewrite all the gui logic as I didn't understand it.
- Other fixes too, happy to elaborate. Works for what I'm doing.

### CODE REVIEW

File: info.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {
 	"name":"CircuitHUD-V2",
-	"version": "1.3.0",
+	"version": "2.0.0",
 	"title":"Circuit HUD V2",
 	"author":"Xiomax & feupeu",
```
Comment (JasonLandbridge) [inline info.json:5]:
```suggestion
	"author":"Xiomax & feupeu",
```
Please add your name here as you also deserve it!

**Comment by JasonLandbridge** [comment]
Thanks! I swear I lost all my Factorio modding/lua skills after 3 years of not working with it. So if you can test it out in various situations without crashing then I will believe you and publish it

**Comment by mGuv** [comment]
I just need to fix a little bit more, but working on it!

**Comment by mGuv** [comment]
Fixed a few more issues but now discovered it's totally broken when you load a game. The HUD is out of sync/frozen, clicking a combinator on the ground crashes the game, etc.

I don't know Factorio's event chain/mod system so unsure how to fix. I've hit a brick wall :(

**Comment by hgschmie** [comment]
I have a full working version (I use it locally) in #7. Feel free to pick that up.

**Pull request closed without merging**